projects/kit/components/input-date/input-date.directive.ts

```diff
@@ -179,6 +152,35 @@ export class TuiInputDateDirective extends TuiInputDateBase<TuiDay> {
         TUI_ITEMS_HANDLERS,
     ).identityMatcher.set((a, b) => a.daySame(b));
 
+    protected readonly mask = tuiMaskito(
+        computed(() =>
+            maskitoDateOptionsGenerator({
+                separator: this.format().separator,
+                mode: TUI_DATE_ADAPTER[this.format().mode],
+                min: this.min().toLocalNativeDate(),
+                max: this.max().toLocalNativeDate(),
+            }),
+        ),
+    );
+
+    protected readonly valueEffect = effect(() => {
+        const value =
+            (this.value() && this.stringify(this.value())) ??
```

Comment:
Unnecessary check for this.value(), stringify already handles null

Reply:
Fixed